File: Facultate/Anul 2/Sem 2/MPP/Laburi/Lab5/CSharpTema3/CSharpLab3/factories/ServiceFactory.cs
using CSharpLab3.repos;
using CSharpLab3.services;
using CSharpLab3.utils;

namespace CSharpLab3.factories;

public class ServiceFactory
{
    private static ServiceFactory? _instance;
    private readonly DbUtils _dbUtils;
    private ServiceFactory()
    {
        _dbUtils = new DbUtils("db.properties");
    }

    public static ServiceFactory Instance => _instance ??= new ServiceFactory();

    public AuthService GetAuthService() =>
        new(new EmployeeDbRepo(_dbUtils));

    public MainService GetMainService() =>
        new(new EmployeeDbRepo(_dbUtils),
            new RidesDbRepo(_dbUtils),
            new ReservationsDbRepo(_dbUtils));
}

File: Facultate/Anul 2/Sem 2/MPP/Laburi/JavaTema4/JavaTema4/Server/src/main/java/utils/ServiceFactory.java
package utils;

import repos.EmployeeDBRepo;
import repos.ReservationsDBRepo;
import repos.RidesDBRepo;
import services.AuthService;
import services.MainService;


public class ServiceFactory {

     private static ServiceFactory instance;

     private ServiceFactory() {

     }

     public static ServiceFactory getInstance() {
          if (instance == null) {
               instance = new ServiceFactory();
          }
          return instance;
     }

     public AuthService getAuthService() {
         return new AuthService(new EmployeeDBRepo(Props.getProperties()));
     }

     public MainService getMainService() {
            return new MainService(new EmployeeDBRepo(Props.getProperties()), new RidesDBRepo(Props.getProperties()), new ReservationsDBRepo(Props.getProperties()));
     }

}
